src/group_theory/solvable.lean

```diff
@@ -157,6 +157,55 @@ general_commutator_eq_commutator G
 
 end derived_series
 
+section commutator_map
+
+variables {G} {G' : Type*} [group G'] {f : G →* G'}
+
+lemma map_commutator_eq_commutator_map (H₁ H₂ : subgroup G) :
+  ⁅H₁, H₂⁆.map f = ⁅H₁.map f, H₂.map f⁆ :=
+begin
+  rw [general_commutator, general_commutator, monoid_hom.map_closure],
+  apply le_antisymm; apply closure_mono,
+  { rintros _ ⟨x, ⟨p, hp, q, hq, rfl⟩, rfl⟩,
+    refine ⟨f p, mem_map.mpr ⟨p, hp, rfl⟩, f q, mem_map.mpr ⟨q, hq, rfl⟩, by simp *⟩, },
+  { rintros x ⟨_, ⟨p, hp, rfl⟩, _, ⟨q, hq, rfl⟩, rfl⟩,
+    refine ⟨p * q * p⁻¹ * q⁻¹, ⟨p, hp, q, hq, rfl⟩, by simp *⟩, },
+end
+
+lemma commutator_le_map_commutator {H₁ H₂ : subgroup G} {K₁ K₂ : subgroup G'} (h₁ : K₁ ≤ H₁.map f)
+  (h₂ : K₂ ≤ H₂.map f) : ⁅K₁, K₂⁆ ≤ ⁅H₁, H₂⁆.map f :=
+by { rw map_commutator_eq_commutator_map, exact general_commutator_mono h₁ h₂ }
+
+section derived_series_map
+
+variables (f)
+
+lemma map_derived_series_le_derived_series (n : ℕ) :
+  (derived_series G n).map f ≤ derived_series G' n :=
+begin
+  induction n with n ih,
+  { simp only [derived_series_zero, le_top], },
+  { simp only [derived_series_succ, map_commutator_eq_commutator_map, general_commutator_mono, *], }
+end
+
+variables {f}
+
+lemma derived_series_le_map_derived_series (hf : function.surjective f) (n : ℕ) :
+  derived_series G' n ≤ (derived_series G n).map f :=
+begin
+  induction n with n ih,
+  { rwa [derived_series_zero, derived_series_zero, top_le_iff, ← monoid_hom.range_eq_map,
+    ← monoid_hom.range_top_iff_surjective.mpr], },
+  { simp only [*, derived_series_succ, commutator_le_map_commutator], }
+end
+
+lemma derived_series_eq_map_derived_series (hf : function.surjective f) (n : ℕ) :
+  derived_series G' n = (derived_series G n).map f :=
```

Comment:
You may consider swapping the two sides of this equality, to put the simplest side on the RHS.

Reply:
Done.